src/components/catalog-search/catalog-search.controller.ts

```diff
@@ -26,27 +25,9 @@ export class CatalogSearchController implements angular.IController {
   }
 
   public $onChanges(onChangesObj: angular.IOnChangesObject) {
-    if (onChangesObj.serviceClasses || onChangesObj.imageStreams) {
-      if (!this.ctrl.serviceClasses && !this.ctrl.imageStreams) {
-        return;
-      }
-
-      // Convert service classes to ServiceItem, which is needed for the ordering panel.
-      let items = _.map(this.ctrl.serviceClasses, (serviceClass) => {
-        return this.Catalog.getServiceItem(serviceClass);
-      });
-
-      // Convert builders to ImageItem.
-      items = items.concat(_.map(this.ctrl.imageStreams, (imageStream) => {
-        return this.Catalog.getImageItem(imageStream);
-      }));
-
-      // Remove null items (non-builder images).
-      items = _.reject(items, (item) => {
-        return !item;
-      });
+    if (onChangesObj.catalogServiceItems && (!onChangesObj.catalogServiceItems.isFirstChange() ||
+      !_.isEmpty(onChangesObj.catalogServiceItems.currentValue))) {
```

Comment:
Fixed, although not sure how/if we want to display an error msg when one occurs.